static/app/types/core.tsx

```diff
@@ -47,8 +48,14 @@ export type Writable<T> = {-readonly [K in keyof T]: T[K]};
 export type SelectValue<T> = {
   label: string | number | React.ReactElement;
   value: T;
+  details?: React.ReactNode;
```

Comment:
Hmmmmm, shouldn't all these things be part of the label? This change seems a bit suspect.

Reply:
I added `details` as a prop so its content can be rendered inside [a styled `<Details>` element](https://github.com/getsentry/sentry/blob/vl/select-option-type/static/app/components/forms/selectOption.tsx#L55), so they would automatically have smaller and lighter text.

<img width="338" alt="Screen Shot 2022-05-05 at 2 17 55 PM" src="https://user-images.githubusercontent.com/44172267/167027022-6e6e30aa-c5ca-46e2-a4db-68ad5f38b4ff.png">

It's just useful for maintaining visual consistency between selectors. If you want to add supplementary but non-essential text, just put it in the `details` prop and don't worry about styling it.